keps/sig-apps/4026-crojob-scheduled-timestamp-annotation/README.md

```diff
@@ -0,0 +1,289 @@
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The CronJob controller will begin adding the scheduled timestamp as an annotation to jobs created while the feature is enabled, and existing jobs will be unaffected.
+
+###### Are there any tests for feature enablement/disablement?
+
+We plan to add unit tests.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+This change will not impact the rollout or rollback fail. It also will not impact the already running workloads.
+
+###### What specific metrics should inform a rollback?
+
+- Users can monitor CronJobs metrics `job_creation_skew_duration_seconds` and `cronjob_controller_rate_limiter_use`, `cronjob_job_creation_skew`.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+The feature will be tested manually prior to beta launch.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+###### How can an operator determine if the feature is in use by workloads?
+
+Randomly checking the CronJobs annotation `batch.kubernetes.io/cronjob-scheduled-timestamp` is sufficient. For monitoring purposes, we can rely on pre-existing metrics which monitor both the cronjob queue and the job creation skew, which should provide sufficient signal if the controller is working as expected. For small clusters, checking  the annotation will determine the feature is used.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [ ] Events
+  - Event Reason:
+- [X] API .metadata
+  - Condition name:
+  - Other field:
+    - `.metadata.annotations['batch.kubernetes.io/cronjob-scheduled-timestamp']`
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+- 99% percentile over day for Job syncs is <= 15s for a client-side 50 QPS limit.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [X] Metrics
+  - Metric name:
+  - `cronjob_job_creation_skew`.
+  - `cronjob_controller_rate_limiter_use`.
+  - `job_creation_skew_duration_seconds`.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+No.
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+New job annotation of size 34B plus value of size N where N is the number of digits in the job.
```

Comment:
I still don't understand this - how is "number of digits in the job" related here.

The job annotation will always contain a timestamp (which is what is what? sth about 25B right?)
[And the name of annotation is 49B if I'm counting correctly]